crypto/build.info

```diff
@@ -1,3 +1,11 @@
+SUBDIRS=objects buffer bio stack lhash rand evp asn1 pem x509 x509v3 conf \
+        txt_db pkcs7 pkcs12 ui kdf store \
+{- my @sdirs = qw(md2 md4 md5 sha mdc2 hmac ripemd whrlpool poly1305 blake2
```

Comment:
I don't think it makes sense to push this sort of perl usage in here - you should handle knocking out directories in the main perl code - not down at this level IMHO.

Reply:
That means we need to list these directories in two places, here and in Configure. That increases the burden of keeping things in sync.

Reply:
No it doesn't - you already filter out the list in Configure itself - you don't build a smaller list. 
You just have to treat the list coming from the build.info files in the same manner. It builds a full list from the build.info files and then you filter it out.
It just means reorganising the logic to handle that.
These should be in the build.info file - but the -no-X processing should remain in the main perl code.
The objective should be to have absolutely no actual perl hacks in the build.info files IMHO.

Reply:
So basically, you want something like this early in the `build.info` lookup loop:

``` perl
        continue if scalar @curd == 2 && $curd[0] eq 'crypto' && $disabled{$curd[1]};
```

and then `crypto/build.info` would have this:

```
SUBDIRS=objects buffer bio stack lhash rand evp asn1 pem x509 x509v3 conf \
        txt_db pkcs7 pkcs12 ui kdf store \
        md2 md4 md5 sha mdc2 hmac ripemd whrlpool poly1305 blake2 \
        siphash sm3 des aes rc2 rc4 rc5 idea aria bf cast camellia \
        seed sm4 chacha modes bn ec rsa dsa dh sm2 dso engine \
        err comp ocsp cms ts srp cmac ct async
```

Do I get that right?

An issue that I see with this is that someone looking at `crypto/build.info` won't see the connection between disabling certain keywords and that certain directories aren't looked into at all, and will have to dig deep in `Configure` to find out (this is the case with the current `sdirs` list as well).  That's confusing for those who do not know the inner workings of Configure.

Reply:
A comment in the build.info pointing to Configure might help.  Very few do understand the inner workings of Configure these days.

Reply:
Ok, I'm listening...  I recalled that there is more information processing in Configure than just the filtering itself.

So yeah, I'll make the suggested changes.